scheduler/server/load_based_sched_alg.go

```diff
@@ -0,0 +1,780 @@
+package server
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/twitter/scoot/common/stats"
+)
+
+const (
+	under int = iota
+	over
+)
+
+// defaults for the LoadBasedScheduler algorithm: only one class and all jobs map to that class
+var DefaultLoadBasedSchedulerClassPercents = map[string]int32{"c0": 100}
+var DefaultRequestorToClassMap = map[string]string{".*": "c0"}
+var DefaultMinRebalanceTime = time.Duration(4 * time.Minute)
+var MaxTaskDuration = time.Duration(4 * time.Hour)
+
+type LoadBasedAlgConfig struct {
+	classLoadPercents           map[string]int
+	classLoadPercentsMu         sync.RWMutex
+	requestorReToClassMap   map[string]string
+	requestorReToClassMapMU sync.RWMutex
+
+	rebalanceThreshold     int
+	rebalanceThresholdMu   sync.RWMutex
+	rebalanceMinDuration   time.Duration
+	rebalanceMinDurationMu sync.RWMutex
+
+	classByDescLoadPct []string
+
+	stat stats.StatsReceiver
+}
+
+// LoadBasedAlg the scheduling algorithm allocates job tasks to workers using a class map as
```

Comment:
In the README, you say that workers are allocated to tasks and it is flipped here. So which is it?

Reply:
rewrote comments/README